executor/analyze.go

```diff
@@ -1321,21 +1365,29 @@ func (e *AnalyzeColumnsExec) subMergeWorker(resultCh chan<- *samplingMergeResult
 	for i := 0; i < l; i++ {
 		retCollector.Base().FMSketches = append(retCollector.Base().FMSketches, statistics.NewFMSketch(maxSketchSize))
 	}
+	e.memTracker.Consume(retCollector.Base().MemSize)
 	for {
 		data, ok := <-taskCh
 		if !ok {
 			break
 		}
+		dataSize := int64(cap(data))
 		colResp := &tipb.AnalyzeColumnsResp{}
 		err := colResp.Unmarshal(data)
 		if err != nil {
 			resultCh <- &samplingMergeResult{err: err}
 			return
 		}
+		colRespSize := int64(colResp.Size())
+		e.memTracker.Consume(colRespSize)
 		subCollector := statistics.NewRowSampleCollector(int(e.analyzePB.ColReq.SampleSize), e.analyzePB.ColReq.GetSampleRate(), l)
-		subCollector.Base().FromProto(colResp.RowCollector)
+		tmpMemSize := subCollector.Base().FromProto(colResp.RowCollector, e.memTracker)
 		UpdateAnalyzeJob(e.ctx, e.job, subCollector.Base().Count)
+		oldRetCollectorSize := retCollector.Base().MemSize
 		retCollector.MergeCollector(subCollector)
+		newRetCollectorSize := retCollector.Base().MemSize
+		subCollectorSize := subCollector.Base().MemSize
+		e.memTracker.Consume(newRetCollectorSize - dataSize - colRespSize - tmpMemSize - oldRetCollectorSize - subCollectorSize)
```

Comment:
I think there can be simplified into `e.memTracker.Consume(rootCollectorSize - oldRootCollectorSize - datasize)

Reply:
Why? What about colRespSize and tmpMemSize? Not small enough to ignore.